pkg/operator/controller/extension/extension/reconciler_delete.go

```diff
@@ -61,7 +66,13 @@ func (r *Reconciler) delete(
 		Log: log,
 	}); err != nil {
 		conditions.installed = v1beta1helper.UpdatedConditionWithClock(r.Clock, conditions.installed, gardencorev1beta1.ConditionFalse, ReasonDeleteFailed, err.Error())
-		return reconcile.Result{}, errors.Join(err, r.updateExtensionStatus(ctx, log, extension, conditions))
+		if updateErr := r.updateExtensionStatus(ctx, log, extension, conditions); updateErr != nil {
+			return reconcile.Result{}, errors.Join(err, fmt.Errorf("failed to update extension status: %w", updateErr))
+		}
+		if !reflect.DeepEqual(reconcileResult, reconcile.Result{}) {
+			return reconcileResult, nil
+		}
+		return reconcile.Result{}, err
```

Comment:
Can you explain this handling, i.e. why does the code ignore errors from all flow steps except "Handling Extension in runtime cluster"?